api/descriptions/collections/name_put.md

```diff
@@ -1,26 +1,62 @@
-Updates the parameters that define a given collection.
+Updates the parameters for a specific collection.
 
-The following example curl command updates the parameters that define the collection named `finance_group_app`.
-In general, all parameters in your PUT request should be defined or redefined.
-Any field left unspecified is assigned the value of `""` (an empty string).
+To invoke this endpoint in the Console UI:
+
+1. Navigate to **Manage > Collections and Tags > Collections**.
+2. Click the dotted icon under the **Actions** column to open up the menu options. **Note:** The default collections do not have a dotted icon in the **Actions** column. Use the  **Manage** cog icon to open the update window.
+3. Click the **Manage** button and update the collection's parameters. 
+4. Click the **Save** button to save the changes.
+
+**Note:** You must define or redefine all parameters in the PUT request. Any parameter left unspecified is set to an empty string.
```

Comment:
@KICKdesigns I played around with the PUT endpoint some more. Although it's true that some unspecified params might be set to an empty string, it's not true for all parameters. For example, unspecified resources (e.g. `hosts`, `images`, `functions`, `clusters` etc seem to be set to a wildcard, which is also the default value for resources in the POST endpoint (see my most recent commit to this PR).

Other params seem mandatory, such as `name`.

Other params seem to get reset, such as `color`.

I think something like `description` gets reset to an empty string.

Can you please play around with this endpoint and better characterize what needs to be specified in the PUT request body and provide this guidance in the documentation? For example, to update the image in `my-collection` seems to be pretty simple if you don't care about retaining the color.

```
curl 'https://<CONSOLE>/api/v1/collections' \
-k \
-X POST \
-u <USER> \
-H 'Content-Type: application/json' \
-d \
'{
 "name":"my-collection",
 "images":["ubuntu:20.04"]
}'
```

Reply:
@iansk I’ve done a few tests with the PUT endpoint:

1. Included everything in payload except color and description. Result: No collection elements changed, description remained as is, color changed to new color.
2. Included only name and owner in the payload. Result: color changed to new random color, no description change, no collections changed, owner did not change.
3. Similar to test 2 but I also included “system” field with changed value. Result: system value not changed.

Based on this it seems like the color changes randomly if not included but everything else remains as is. I've updated the doc to reflect this. The original note (from the original doc) was incorrect and thus it has been deleted in this revision.

Reply:
@KICKdesigns Thanks for testing this endpoint and summarizing the results here.

Can you please take some of these learnings and add them as "usage" notes? Since we don't specify required and optional for request params, it's important that we help get customers come up to speed through the examples and descriptions.

 In this case, I see the usage notes being something like this:

* Name is mandatory
* List of what's considered to be a "resource".
* If a resource is left unspecified, its default values is a wildcard.
* If `color` is left unspecified, the system picks a random value for you.
* If `description` is left unspecified, the default value is an empty string.

Your findings for `owner` and `system` make sense. They work similarly for all endpoints. Also, `modified` works the same for all endpoints. It never needs to be specified. The system will update it for you automatically.

Also, your curl command for updating the image can be further simplified. Assuming you don't care about color or description and given that the starting state for all resources except `images` is a wildcard, the following curl command is sufficient:

```
curl 'https://<CONSOLE>/api/v1/collections' \
-k \
-X POST \
-u <USER> \
-H 'Content-Type: application/json' \
-d \
'{
 "name":"my-collection",
 "images":["ubuntu:20.04"]
}'
```